freeipa/src/test/java/com/sequenceiq/freeipa/service/upgrade/ccm/UpgradeCcmOrchestratorServiceTest.java

```diff
@@ -0,0 +1,79 @@
+package com.sequenceiq.freeipa.service.upgrade.ccm;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.sequenceiq.cloudbreak.orchestrator.exception.CloudbreakOrchestratorException;
+import com.sequenceiq.cloudbreak.orchestrator.host.HostOrchestrator;
+import com.sequenceiq.cloudbreak.orchestrator.host.OrchestratorStateParams;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.service.GatewayConfigService;
+import com.sequenceiq.freeipa.service.freeipa.flow.FreeIpaNodeUtilService;
+import com.sequenceiq.freeipa.service.stack.StackService;
+
+@ExtendWith(MockitoExtension.class)
+class UpgradeCcmOrchestratorServiceTest {
+
+    private static final long STACK_ID = 123L;
+
+    @Mock
+    private StackService stackService;
+
+    @Mock
+    private GatewayConfigService gatewayConfigService;
+
+    @Mock
+    private FreeIpaNodeUtilService freeIpaNodeUtilService;
+
+    @Mock
+    private HostOrchestrator hostOrchestrator;
+
+    @Mock
+    private Stack stack;
+
+    @Captor
+    private ArgumentCaptor<OrchestratorStateParams> paramCaptor;
+
+    @InjectMocks
+    private UpgradeCcmOrchestratorService underTest;
+
+    @BeforeEach
+    void setUp() {
+        when(stackService.getByIdWithListsInTransaction(STACK_ID)).thenReturn(stack);
+    }
+
+    @Test
+    void testApplyUpgradeState() throws CloudbreakOrchestratorException {
+        underTest.applyUpgradeState(STACK_ID);
+        verify(hostOrchestrator).runOrchestratorState(paramCaptor.capture());
+        OrchestratorStateParams params = paramCaptor.getValue();
+        assertThat(params.getState()).isEqualTo("upgradeccm");
+    }
+
+    @Test
+    void testReconfigureNginx() throws CloudbreakOrchestratorException {
+        underTest.reconfigureNginx(STACK_ID);
+        verify(hostOrchestrator).runOrchestratorState(paramCaptor.capture());
+        OrchestratorStateParams params = paramCaptor.getValue();
+        assertThat(params.getState()).isEqualTo("nginx");
+    }
+
+    @Test
+    void testDisableMina() throws CloudbreakOrchestratorException {
+        underTest.disableMina(STACK_ID);
+        verify(hostOrchestrator).runOrchestratorState(paramCaptor.capture());
+        OrchestratorStateParams params = paramCaptor.getValue();
+        assertThat(params.getState()).isEqualTo("upgradeccm/disable-ccmv1");
+    }
```

Comment:
could you validate the others field too?

Reply:
sure